internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -285,6 +302,29 @@ func resolveCTPolicyTargetRef(policy *egv1a1.ClientTrafficPolicy, gateways []*Ga
 	return gateway
 }
 
+func validatePortOverlapForClientTrafficPolicy(l *ListenerContext, xds *ir.Xds) error {
```

Comment:
can this be also shared with SecurityPolicy and moved into helpers.go ?

Reply:
Not easily. CTP is listener oriented, while SP is route oriented. The way that the correct listener is found is different in each of these validation methods.